WcaOnRails/app/views/results/_rankings_selector.html.erb

```diff
@@ -0,0 +1,144 @@
+          <% else %>
+            <%= t("results.rankings.form_elements.years_selector.until") %>
+          <% end %>
+        </span>
+      </label>
+    </div>
+  </div>
+</div>
+
+<div class="form-group">
+  <%= label_tag(t("results.rankings.form_elements.show_selector.show")) %>
+  <div id="show" class="btn-group">
+    <div class="btn-group">
+      <% persons_active = @is_persons %>
+      <label id="persons" class="btn btn-info <%= persons_active ? " active" : "" %>" data-toggle="dropdown">
+        <span class="caption">
+          <% if persons_active %>
+            <%= t("results.rankings.form_elements.show_selector.persons_html", quantity: @show) %>
+          <% else %>
+            <%= t("results.rankings.form_elements.show_selector.persons") %>
+          <% end %>
+        </span>
+      </label>
+      <ul class="dropdown-menu show-persons">
+        <% @quantities.each do |quantity| %>
+          <li class="persons <%= (persons_active && quantity.to_s == @show ? " active" : "") %>"><%= link_to quantity.to_s, request.params.merge(show: "#{quantity} persons") %></li>
+        <% end %>
+      </ul>
+    </div>
+    <div class="btn-group">
+      <% results_active = @is_results %>
+      <label id="results" class="btn btn-info <%= results_active ? " active" : "" %>" data-toggle="dropdown">
+        <span class="caption">
+          <% if results_active %>
+            <%= t("results.rankings.form_elements.show_selector.results_html", quantity: @show) %>
+          <% else %>
+            <%= t("results.rankings.form_elements.show_selector.results") %>
+          <% end %>
+        </span>
+      </label>
+      <ul class="dropdown-menu show-results">
+        <% @quantities.each do |quantity| %>
+          <li class="results <%= (results_active && quantity.to_s == @show ? " active" : "") %>"><%= link_to quantity.to_s, request.params.merge(show: "#{quantity} results") %></li>
+        <% end %>
+      </ul>
+    </div>
+    <%= link_to t("results.rankings.form_elements.show_selector.by_region"), request.params.merge(show: "by region"), class: "btn btn-info" + (@is_by_region ? " active" : "") %>
+  </div>
+</div>
+
+<script>
+  $(function() {
+    $('#region').bind('change', function () {
+      var url = window.location.href;
+      var hash = location.hash;
+      url = url.replace(hash, '');
+      if (url.indexOf("region=") >= 0) {
+        var prefix = url.substring(0, url.indexOf("region="));
+        var suffix = url.substring(url.indexOf("region="));
+        suffix = suffix.substring(suffix.indexOf("=") + 1);
+        suffix = (suffix.indexOf("&") >= 0) ? suffix.substring(suffix.indexOf("&")) : "";
+        url = prefix + "region=" + $(this).val() + suffix;
+      } else {
+        if (url.indexOf("?") < 0) {
+          url += "?" + "region=" + $(this).val();
+        } else {
+          url += "&" + "region=" + $(this).val();
+        }
+      }
+      window.location.href = url + hash;
+    });
```

Comment:
I believe we have some jQuery plugins for managing params, this could be something like:
```js
var params = $.getUrlParams();
params.region = $(this).val();
window.location.search = $.param(params);
```

Reply:
That's so much better, I've updated it already! :slightly_smiling_face: